### PR TITLE
python3Packages.sslyze: 3.0.0 -> 3.1.0

### PR DESCRIPTION
**Pull request by veehaitch**
###### Motivation for this change

Update to version 3.1.0

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/sslyze/default.nix [hunk 1]
```diff
@@ -10,20 +10,15 @@
...
-    sha256 = "06mwzxw6xaqin2gwzcqb9r7qhbyx3k7zcxygxywi2bpxyjv9lq32";
+    sha256 = "02p0lgpkfq88dys0dqw0z8bpg9g8pds2lvs9awd9f2w5cb1pwr83";
   };
 
-  patchPhase = ''
-    substituteInPlace setup.py \
-      --replace "cryptography>=2.6,<=2.9" "cryptography"
-  '';
-
   checkInputs = [ pytest ];
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/sslyze/default.nix:22]: Please use pytestCheckHook.

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 105905` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>python39Packages.sslyze</li>
  </ul>
</details>
<details>
  <summary>5 packages built:</summary>
  <ul>
    <li>python37Packages.nassl</li>
    <li>python37Packages.sslyze</li>
    <li>python38Packages.nassl</li>
    <li>sslyze (python38Packages.sslyze)</li>
    <li>python39Packages.nassl</li>
  </ul>
</details>

File: pkgs/development/python-modules/sslyze/default.nix (hunk 1 above)
Comment (veehaitch) [reply pkgs/development/python-modules/sslyze/default.nix]: Sure, done!

**Comment by veehaitch** [comment]
> Result of `nixpkgs-review pr 105905` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
> 1 package failed to build:
> 5 packages built:

A test for `random2` (`sslyze` -> `faker` -> `random2`) fails due to [a change introduced in Python 3.9](https://docs.python.org/3/library/random.html#random.getrandbits). I added a commit which patches `random2` accordingly. Should be fine now.

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 105905` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>14 packages marked as broken and skipped:</summary>
  <ul>
    <li>gscan2pdf</li>
    <li>mailman-web</li>
    <li>odpdown</li>
    <li>pysolfc</li>
    <li>python37Packages.mailman-web</li>
    <li>python37Packages.qasm2image</li>
    <li>python37Packages.rl-coach</li>
    <li>python37Packages.toggl-cli</li>
    <li>python38Packages.mailman-web</li>
    <li>python38Packages.qasm2image</li>
    <li>python38Packages.toggl-cli</li>
    <li>python39Packages.mailman-web</li>
    <li>python39Packages.qasm2image</li>
    <li>python39Packages.toggl-cli</li>
  </ul>
</details>
<details>
  <summary>38 packages failed to build:</summary>
  <ul>
    <li>prospector</li>
    <li>python37Packages.django_extensions</li>
    <li>python37Packages.django_silk</li>
    <li>python37Packages.nassl</li>
    <li>python37Packages.pylint-django</li>
    <li>python37Packages.qiskit</li>
    <li>python37Packages.qiskit-aqua</li>
    <li>python37Packages.scrapy-fake-useragent</li>
    <li>python37Packages.sslyze</li>
    <li>python38Packages.django_extensions</li>
    <li>python38Packages.django_silk</li>
    <li>python38Packages.nassl</li>
    <li>python38Packages.pylint-django</li>
    <li>python38Packages.qiskit</li>
    <li>python38Packages.qiskit-aqua</li>
    <li>python38Packages.scrapy-fake-useragent</li>
    <li>sslyze (python38Packages.sslyze)</li>
    <li>python39Packages.Quandl (python39Packages.quandl)</li>
    <li>python39Packages.afdko</li>
    <li>python39Packages.booleanoperations</li>
    <li>python39Packages.defcon</li>
    <li>python39Packages.django_extensions</li>
    <li>python39Packages.django_silk</li>
    <li>python39Packages.fontmath</li>
    <li>python39Packages.fontparts</li>
    <li>python39Packages.fontpens</li>
    <li>python39Packages.fonttools</li>
    <li>python39Packages.mutatormath</li>
    <li>python39Packages.nassl</li>
    <li>python39Packages.nototools</li>
    <li>python39Packages.psautohint</li>
    <li>python39Packages.pylint-django</li>
    <li>python39Packages.qiskit</li>
    <li>python39Packages.qiskit-aqua</li>
    <li>python39Packages.sslyze</li>
    <li>python39Packages.ufoprocessor</li>
    <li>sourcehut.gitsrht</li>
    <li>sourcehut.todosrht</li>
  </ul>
</details>
<details>
  <summary>58 packages built:</summary>
  <ul>
    <li>flexget</li>
    <li>freecell-solver</li>
    <li>noto-fonts-emoji</li>
    <li>photon</li>
    <li>python37Packages.Quandl (python37Packages.quandl)</li>
    <li>python37Packages.afdko</li>
    <li>python37Packages.booleanoperations</li>
    <li>python37Packages.defcon</li>
    <li>python37Packages.factory_boy</li>
    <li>python37Packages.faker</li>
    <li>python37Packages.flask-restx</li>
    <li>python37Packages.fontmath</li>
    <li>python37Packages.fontparts</li>
    <li>python37Packages.fontpens</li>
    <li>python37Packages.fonttools</li>
    <li>python37Packages.minio</li>
    <li>python37Packages.mutatormath</li>
    <li>python37Packages.nototools</li>
    <li>python37Packages.psautohint</li>
    <li>python37Packages.pytest-factoryboy</li>
    <li>python37Packages.pytest-randomly</li>
    <li>python37Packages.random2</li>
    <li>python37Packages.tld</li>
    <li>python37Packages.ufoprocessor</li>
    <li>python38Packages.Quandl (python38Packages.quandl)</li>
    <li>python38Packages.afdko</li>
    <li>python38Packages.booleanoperations</li>
    <li>python38Packages.defcon</li>
    <li>python38Packages.factory_boy</li>
    <li>python38Packages.faker</li>
    <li>python38Packages.flask-restx</li>
    <li>python38Packages.fontmath</li>
    <li>python38Packages.fontparts</li>
    <li>python38Packages.fontpens</li>
    <li>python38Packages.fonttools</li>
    <li>python38Packages.minio</li>
    <li>python38Packages.mutatormath</li>
    <li>python38Packages.nototools</li>
    <li>python38Packages.psautohint</li>
    <li>python38Packages.pytest-factoryboy</li>
    <li>python38Packages.pytest-randomly</li>
    <li>python38Packages.random2</li>
    <li>python38Packages.tld</li>
    <li>python38Packages.ufoprocessor</li>
    <li>python39Packages.factory_boy</li>
    <li>python39Packages.faker</li>
    <li>python39Packages.flask-restx</li>
    <li>python39Packages.minio</li>
    <li>python39Packages.pytest-factoryboy</li>
    <li>python39Packages.pytest-randomly</li>
    <li>python39Packages.random2</li>
    <li>python39Packages.scrapy-fake-useragent</li>
    <li>python39Packages.tld</li>
    <li>rictydiminished-with-firacode</li>
    <li>scfbuild</li>
    <li>visidata</li>
    <li>xits-math</li>
    <li>xquartz</li>
  </ul>
</details>

**Comment by veehaitch** [comment]
> Result of `nixpkgs-review pr 105905` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)

The packages this PR relates to seem to build fine for me. I doubt the failing packages have anything to do with the changes this PR introduces.

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 105905` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages failed to build:</summary>
  <ul>
    <li>python37Packages.sslyze</li>
    <li>sslyze (python38Packages.sslyze)</li>
    <li>python39Packages.sslyze</li>
  </ul>
</details>
<details>
  <summary>10 packages built:</summary>
  <ul>
    <li>freecell-solver</li>
    <li>gnome3.aisleriot</li>
    <li>kdeApplications.kpat</li>
    <li>pysolfc</li>
    <li>python37Packages.nassl</li>
    <li>python37Packages.random2</li>
    <li>python38Packages.nassl</li>
    <li>python38Packages.random2</li>
    <li>python39Packages.nassl</li>
    <li>python39Packages.random2</li>
  </ul>
</details>

sslyze:

```
ERROR: Could not find a version that satisfies the requirement cryptography<3.3,>=2.6 (from sslyze)
ERROR: No matching distribution found for cryptography<3.3,>=2.6
```

**Comment by veehaitch** [comment]
> Result of `nixpkgs-review pr 105905` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
> 3 packages failed to build:
> 10 packages built:
> 
> sslyze:
> 
> ```
> ERROR: Could not find a version that satisfies the requirement cryptography<3.3,>=2.6 (from sslyze)
> ERROR: No matching distribution found for cryptography<3.3,>=2.6
> ```

Thanks, fixed!

**Comment by veehaitch** [comment]
`nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
> 6 packages built:
python37Packages.nassl python37Packages.sslyze python38Packages.nassl sslyze python39Packages.nassl python39Packages.sslyze

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review which does not build all packages (e.g. lumo, tensorflow or pytorch)
If you find some bugs or got suggestions for further things to search or run please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 105905` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>

  <summary>7 packages failed to build and are new build failures:</summary>
  <ul>
<li>python37Packages.nassl: log <a href="https://termbin.com/u9k0">https://termbin.com/u9k0</a></li>
<li>python37Packages.sslyze: log was empty</li>
<li>python38Packages.nassl: log <a href="https://termbin.com/6u9i">https://termbin.com/6u9i</a></li>
<li>sslyze: log was empty</li>
<li>(python38Packages.sslyze): log was empty</li>
<li>python39Packages.nassl: log <a href="https://termbin.com/zimo">https://termbin.com/zimo</a></li>
<li>python39Packages.sslyze: log was empty</li>
  </ul>
</details>

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review which does not build all packages (e.g. lumo, tensorflow or pytorch)
If you find some bugs or got suggestions for further things to search or run please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 105905` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
6 packages built:
</summary>
<ul>
<li>python37Packages.nassl</li>
<li>python37Packages.sslyze</li>
<li>python38Packages.nassl</li>
<li>sslyze (python38Packages.sslyze)</li>
<li>python39Packages.nassl</li>
<li>python39Packages.sslyze</li>
</ul>
</details>

**Review by jonringer — APPROVED** [review]
LGTM

Result of nixpkgs-review pr 105905 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)<details>  <summary>6 packages built:</summary>  <ul>    <li>python37Packages.nassl</li>    <li>python37Packages.sslyze</li>    <li>python38Packages.nassl</li>    <li>sslyze (python38Packages.sslyze)</li>    <li>python39Packages.nassl</li>    <li>python39Packages.sslyze</li>  </ul></details>

**Pull request merged**